docs/rules/func-names.md

```diff
@@ -45,6 +46,34 @@ Foo.prototype.bar = function bar() {};
 }())
 ```
 
+### as-needed
+
+In ES6, aside from explicit names, function names may be come from the surrounding context of the function. For example, a function assigned to a variable will automatically use the name of the variable as its name if it have no name of its own.
```

Comment:
Few small typos. How does this sound? `"In ES6, aside from explicit names, function names may come from the surrounding context of the function. For example, a function assigned to a variable will automatically use the name of the variable as its name if it has no name of its own."`